services/cache/lru.go

```diff
@@ -182,6 +196,34 @@ func (l *LRU) get(key string, value interface{}) error {
 
 		l.evictList.MoveToFront(ent)
 
+		// We use a fast path for hot structs.
+		if msgpVal, ok := value.(msgp.Unmarshaler); ok {
+			_, err := msgpVal.UnmarshalMsg(e.value)
+			return err
+		}
+
+		// This is ugly and makes the cache package aware of the model package.
+		// But this is due to 2 things.
+		// 1. The msgp package works on methods on structs rather than functions.
+		// 2. Our cache interface passes pointers to empty pointers, and not pointers
```

Comment:
Pointers to empty pointers? did you mean pointeres to empty interfaces?

Reply:
I meant the first. `var session *Session` and then `&session` is a pointer to a nil pointer. The fact that it eventually gets passed through `interface{}` is different, but not strictly related to the problem.